LFPy/run_simulation.pyx

```diff
@@ -181,25 +194,29 @@ def _run_simulation_with_electrode(cell, cvode, electrode=None,
     if to_memory:
         electrodesLFP = []
         for coeffs in dotprodcoeffs:
-            electrodesLFP.append(np.zeros((coeffs.shape[0],
-                                    int(tstop / dt + 1))))
+            if not cvode.active():
+                # if cvode is on lists are created on-the-fly
+                electrodesLFP.append(np.zeros((coeffs.shape[0],
+                                     int(cell.tstop / cell.dt) + 1)))
     #LFPs for each electrode will be put here during simulations
     if to_file:
         #ensure right ending:
-        if file_name.split('.')[-1] != 'h5':
-            file_name += '.h5'
-        el_LFP_file = h5py.File(file_name, 'w')
-        i = 0
-        for coeffs in dotprodcoeffs:
-            el_LFP_file['electrode{:03d}'.format(i)] = np.zeros((coeffs.shape[0],
-                                            int(tstop / dt + 1)))
-            i += 1
+        file_name = Path(file_name)
+        if file_name.suffix != '.h5':
+            file_name = file_name.parent / (file_name.name + '.h5')
+        if not cvode.active():
+            el_LFP_file = h5py.File(file_name, 'w')
+            i = 0
+            for coeffs in dotprodcoeffs:
+                el_LFP_file['electrode{:03d}'.format(i)] = np.zeros((coeffs.shape[0], int(cell.tstop / cell.dt + 1)))
+                i += 1
 
     # create a 2D array representation of segment midpoints for dot product
     # with transmembrane currents when computing dipole moment
     if rec_current_dipole_moment:
-        current_dipole_moment = cell.current_dipole_moment.copy()
-        cell.current_dipole_moment = np.array([[]])
+        if not cvode.active():
+            current_dipole_moment = cell.current_dipole_moment.copy()
+            cell.current_dipole_moment = np.array([[]])
```

Comment:
This doesn't look consistent with the implementation in the `.py` version. Also, `current_dipole_moment` is C-declared which means that it should be much faster to read/write to in nested for loops. I wonder if it would be worthwhile to buffer the output array for CVode stuff. If the buffer is used up, copy into a new array with bigger buffer, and repeat if necessary. After the final tstep, remove empty rows. What do you think?

Reply:
I can do that

Reply:
But I don't know if it's worth it. Lists are pretty fast anyways.
Also, since dt can be set by the user (any dt can be used) it might be that for big dt the buffer is not large enough. I would keep it this way

Reply:
BTW, this copy was there before. Why is it not consistent with the py implementation?

Reply:
Hi. I see now. The .py version doesn't copy (L193-4 should be equivalent) because there is nothing to gain. Maybe we shall leave optimization out for this PR, but my thinking is that list append in a while loop must be orders of magnitude slower than filling in rows or columns in a C-declared and typed matrix using Cython. But one has to consider that calling fadvance() may take a longer time in many models anyway. L240 below also gains nothing from Cythonizing, and that's stuff I wrote presumably :-/